src/agent/communicator/include/ihttp_client.hpp

```diff
@@ -25,21 +26,22 @@ namespace http_client
 
         HttpRequestParams(boost::beast::http::verb method,
                           const std::string& host,
-                          const std::string& port,
                           const std::string& endpoint,
-                          const bool use_https,
                           const std::string& token = "",
                           const std::string& user_pass = "",
                           const std::string& body = "")
             : Method(method)
             , Host(host)
-            , Port(port)
             , Endpoint(endpoint)
-            , Use_Https(use_https)
             , Token(token)
             , User_pass(user_pass)
             , Body(body)
         {
+            boost::urls::url_view url(host);
```

Comment:
What if it is an invalid url?

Reply:
This case has been handled so there is no crash or exception thrown. A more thorough validation of configuration settings is necessary, though.